Common/Orders/Fills/EquityFillModel.cs

```diff
@@ -242,346 +215,539 @@ public virtual OrderEvent StopLimitFill(Security asset, StopLimitOrder order)
                 return fill;
             }
 
-            //Get the range of prices in the last bar:
-            var prices = GetPricesCheckingPythonWrapper(asset, order.Direction);
-            var pricesEndTime = prices.EndTime.ConvertToUtc(asset.Exchange.TimeZone);
+            // Get the last trade bar since stop limit orders are triggered by trades
+            var lastTradeBar = GetLastTradeBar(asset);
 
-            // do not fill on stale data
-            if (pricesEndTime <= order.Time) return fill;
+            // Do not fill on stale data
+            var localOrderTime = order.Time.ConvertFromUtc(asset.Exchange.TimeZone);
+            if (lastTradeBar.EndTime <= localOrderTime) return fill;
 
-            //Check if the Stop Order was filled: opposite to a limit order
+            // Check if the Stop Order was filled: opposite to a limit order
             switch (order.Direction)
             {
                 case OrderDirection.Buy:
-                    //-> 1.2 Buy Stop: If Price Above Setpoint, Buy:
-                    if (prices.High > order.StopPrice || order.StopTriggered)
+                    //-> 1.1 Buy Stop: If Price Above set point, Buy:
+                    if (lastTradeBar.High >= order.StopPrice || order.StopTriggered)
                     {
                         order.StopTriggered = true;
+                        DateTime unused;
 
                         // Fill the limit order, using closing price of bar:
-                        // Note > Can't use minimum price, because no way to be sure minimum wasn't before the stop triggered.
-                        if (prices.Current < order.LimitPrice)
+                        // Note > Can't use minimum price, because no way to be sure minimum wasn't before the stop triggered,
+                        // so we will will the current ask price
+                        if (GetAskPrice(asset, out unused) <= order.LimitPrice)
                         {
+                            fill.FillPrice = Math.Min(lastTradeBar.High, order.LimitPrice);
```

Comment:
I think a further improvement here could be: in the case `order.StopTriggered` is already true from a previous check, we could get the entire QuoteBar (instead of just using the `.Close` as here) and could use the `.Low`? @jaredbroad @AlexCatarino thoughts?